scripts/nb-tester/qiskit_docs_notebook_tester/config.py

```diff
@@ -59,159 +59,146 @@ def get_notebook_jobs(args: argparse.Namespace) -> Iterator[NotebookJob]:
     """
     config = Config.from_args(args)
 
-    for path in config.notebooks_to_execute():
-        backend_patch = None
-        if config.should_patch(path):
-            # Implements a subset of options from QiskitRuntimeService, but in 
-            # practice any option can easily be added here
-            backend_patch = generate_backend_patch(
-                backend_name=config.args.backend, 
-                provider=config.args.provider,
-                runtime_service_kwargs = {
-                    "channel": config.args.channel,
-                    "token": config.args.token,
-                    "url": config.args.url,
-                    "name": config.args.name,
-                    "instance": config.args.instance,
-                },
-                generic_backend_kwargs = {
-                    "num_qubits": config.args.num_qubits,
-                    "control_flow": config.args.control_flow
-                }
+    for group in config.groups:
+        for path in group["notebooks"]:
+            if config.cli_filenames and not matches(path, config.cli_filenames):
+                continue
+            if config.test_strategy not in group.get("test-strategies", {}):
+                if matches(path, config.cli_filenames):
+                    # User explicitly defined this path so we'll explain why we're skipping it
+                    print(
+                        f'ℹ️ Skipping {path} (no test strategy "{config.test_strategy}" defined for this notebook)'
+                    )
+                continue
+
+            patch = config.get_patch_for_group(group)
+
+            def should_write(config: Config, patch: str | None):
```

Comment:
Why define a function, given that it's only invoked one time?

Reply:
I thought it'd be more readable. Alternative is something like the following.

```python
write = None
if patch:
    write = Result(False, ...)
elif not config.write:
    write = Result(False, ...)
else:
   write = Result(True)
```

We should go with whichever is clearer to you; I've seen the code too much to make a good judgement.

Reply:
Python is really weird with variable scope. You don't need the initial `write = None`, unlike JS and most languages. 
<img width="218" alt="Screenshot 2024-12-03 at 12 50 41 PM" src="https://github.com/user-attachments/assets/c2ac45d6-ec4c-4eeb-a8fe-bd3f330b949d">

I find that style the most readable.